sdk/openai/openai/test/public/utils/utils.ts

```diff
@@ -0,0 +1,232 @@
+    const modelName = deployment.properties?.model?.name;
+    const modelVersion = deployment.properties?.model?.version;
+    if (deploymentName && modelName && modelVersion) {
+      deployments.push({ deploymentName, model: { name: modelName, version: modelVersion } });
+    }
+  }
+  logger.info(`Available deployments (${deployments.length}): ${JSON.stringify(deployments)}`);
+  return deployments;
+}
+
+export function updateWithSucceeded(
+  succeeded: DeploymentInfo[],
+  deployments: DeploymentInfo[],
+): void {
+  if (deployments.length === 0) {
+    deployments.push(...succeeded);
+  }
+}
+
+export function getSucceeded(
+  deployments: DeploymentInfo[],
+  succeededDeployments: DeploymentInfo[],
+): DeploymentInfo[] {
+  if (succeededDeployments.length > 0) {
+    return succeededDeployments;
+  }
+  return deployments;
+}
+
+function getAccountNameFromResourceType(deploymentType: DeploymentType): string {
+  switch (deploymentType) {
+    case "completions":
+      return assertEnvironmentVariable(
+        EnvironmentVariableNamesForCompletions.ACCOUNT_NAME_COMPLETIONS,
+      );
+    case "vision":
+      return assertEnvironmentVariable(EnvironmentVariableNamesForVision.ACCOUNT_NAME_VISION);
+    case "audio":
+      return assertEnvironmentVariable(EnvironmentVariableNamesForAudio.ACCOUNT_NAME_AUDIO);
+  }
+}
+
+export async function getDeployments(deploymentType: DeploymentType): Promise<DeploymentInfo[]> {
+  return listDeployments(
+    assertEnvironmentVariable(EnvironmentVariableNamesAzureCommon.SUBSCRIPTION_ID),
+    assertEnvironmentVariable(EnvironmentVariableNamesAzureCommon.RESOURCE_GROUP),
+    getAccountNameFromResourceType(deploymentType),
+  );
+}
+
+export async function bufferAsyncIterable<T>(iter: AsyncIterable<T>): Promise<T[]> {
+  const result: T[] = [];
+  for await (const item of iter) {
+    result.push(item);
+  }
+  return result;
+}
+
+export async function get(url: string): Promise<PipelineResponse> {
+  const request = createPipelineRequest({
+    url,
+    method: "GET",
+    headers: createHttpHeaders(),
+    streamResponseStatusCodes: new Set([200]),
+  });
+  return sendRequestWithRecorder(request);
+}
+
+export function createAzureSearchExtension(): AzureChatExtensionConfiguration {
+  return {
+    type: "azure_search",
+    parameters: {
+      endpoint: assertEnvironmentVariable(
+        EnvironmentVariableNamesForAzureSearch.AZURE_SEARCH_ENDPOINT,
+      ),
+      index_name: assertEnvironmentVariable(
+        EnvironmentVariableNamesForAzureSearch.AZURE_SEARCH_INDEX,
+      ),
+      authentication: {
+        type: "system_assigned_managed_identity",
+      },
+    },
+  };
+}
+
+export function handleAssistantsRunFailure(
+  run: Run,
+  context: TaskContext<Test> & TestContext & object,
+): void {
+  if (run.status === "failed") {
+    if (run.last_error?.message.includes("Rate limit")) {
+      logger.info(`Rate limit error: ${run.last_error.message}`);
+      context.skip();
```

Comment:
it is a bit overkill to do the skipping from the inside. We could simplify this function to be a predicate that determines if the test should be skipped and then the test can do the skipping.